docs/api.md

```diff
@@ -2971,6 +2980,51 @@ const { chromium } = require('playwright');  // Or 'firefox' or 'webkit'.
 - returns: <[string]> Dialog's type, can be one of `alert`, `beforeunload`, `confirm` or `prompt`.
 
 
+### class: Download
+
+[Download] objects are dispatched by page via the ['download'](#event-download) event.
+
+Note that browser context must be created with the `acceptDownloads` set to `true` when user needs access to the downloaded content. If `acceptDownloads` is not set or set to `false`, download events are emitted, but the actual download is not performed and user has no access to the downloaded files.
+
+All the downloaded files belonging to the browser context are deleted when the browser context is closed. All downloaded files are deleted when the browser closes.
+
+An example of using `Download` class:
+```js
+const [ download ] = await Promise.all([
+  page.waitForEvent('download'),
+  page.click('a')
+]);
+const path = await download.path();
+...
+```
+
+<!-- GEN:toc -->
+- [download.createReadStream()](#downloadcreatereadstream)
+- [download.delete()](#downloaddelete)
+- [download.error()](#downloaderror)
+- [download.path()](#downloadpath)
+<!-- GEN:stop -->
+
+#### download.createReadStream()
+- returns: <[Promise]<null|[stream.Readable]>>
+
+Returns readable stream for current download or `null` if download failed.
+
+#### download.delete()
+- returns: <[Promise]>
+
+Deletes the downloaded file.
+
+#### download.error()
+- returns: <[Promise]<null|[string]>>
```

Comment:
What kind of errors are returned here? When should I use this string?

Reply:
WebKit will report network error here. If you did not pass `acceptDownloads`, you get `canceled` error as well.

Reply:
Can this match the `response.failure()` api? It is equivalent right?

Reply:
It is not equivalent, but it is similar, so i can rename.